x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.tsx

```diff
@@ -109,38 +109,31 @@ export const CredentialsList: React.FC = () => {
     },
   ];
 
-  const pagination = {
-    pageIndex: meta.page ? meta.page.current - 1 : 0,
-    pageSize: meta.page ? meta.page.size : 0,
-    totalItemCount: meta.page ? meta.page.total_results : 0,
-    hidePerPageOptions: true,
-  };
-
-  const onTableChange = ({ page }: CriteriaWithPagination<ApiToken>) => {
-    const { index: current } = page;
-    fetchCredentials(current + 1);
-  };
-
-  return items.length < 1 ? (
-    <EuiEmptyPrompt
-      iconType="editorStrike"
-      title={
-        <h2>
-          {i18n.translate('xpack.enterpriseSearch.appSearch.credentials.empty.title', {
-            defaultMessage: 'No API Keys have been created yet.',
-          })}
-        </h2>
-      }
-      body={i18n.translate('xpack.enterpriseSearch.appSearch.credentials.empty.body', {
-        defaultMessage: 'Click the "Create a key" button to make your first one.',
-      })}
-    />
-  ) : (
+  return (
     <EuiBasicTable
       columns={columns}
       items={items}
-      pagination={pagination}
-      onChange={onTableChange}
+      noItemsMessage={
+        <EuiEmptyPrompt
```

Comment:
I moved the EmptyPrompt to the `noItemsMessage` prop  rather than using conditional JSX. I like this approach because it still gives context clues as to the table columns/data even when there are no items. We're using `noItemsMessage` right now in Analytics tables & the Curations table, so I thought I'd update this table to match as well.

Reply:
I forgot to note this in the EnginesTable comment section, but I did not convert EnginesTable to use `noItemsMessage` in this PR as this may be something @byronhulcher tackles as part of his upcoming engines/meta engines work (as he may have to create an entirely new table component for meta engines)